src/model_service.cpp

```diff
@@ -54,8 +54,8 @@ Status GetModelStatusImpl::createGrpcRequest(std::string model_name, const std::
     request->mutable_model_spec()->set_name(model_name);
     if (model_version.has_value()) {
         if (model_version.value() < 0) {
-            spdlog::error("Version in GetModelStatus request cannot be negative. Provided value {}", model_version.value());
-            return StatusCode::MODEL_VERSION_MISSING;
+            SPDLOG_ERROR("Illegal model version requested:{}. Model version should be a number no less than 0.", model_version.value());
+            return StatusCode::MODEL_VERSION_ILLEGAL_VALUE;
         }
         request->mutable_model_spec()->mutable_version()->set_value(model_version.value());
```

Comment:
What if user requests version `-5` in `GetModelStatusImpl::getModelStatus`?

Reply:
getModelStatus will return MODEL_VERSION_ILLEGAL_VALUE(see GetModelMetadataImpl::validate)

Reply:
This is GetModelStatus, not GetModelMetadata. These are 2 separate endpoints. GetModelStatus does not perform validation.

Reply:
Maybe add such unit test to ensure MODEL_VERSION_ILLEGAL_VALUE will be returned?